### PR TITLE
change deployer permissions

### PR DESCRIPTION
**Pull request by ilinzweilin**

### CODE REVIEW

File: script/Deployer.sol [hunk 1]
```diff
@@ -50,13 +53,17 @@ contract Deployer is Script {
...
+        AuthLike(restrictionManagerFactory).rely(address(poolManager));
+
+        AuthLike(liquidityPoolFactory).rely(address(root));
+        AuthLike(trancheTokenFactory).rely(address(root));
+        AuthLike(restrictionManagerFactory).rely(address(root));
 
-        LiquidityPoolFactory(liquidityPoolFactory).rely(address(root));
-        TrancheTokenFactory(trancheTokenFactory).rely(address(root));
-        RestrictionManagerFactory(restrictionManagerFactory).rely(address(root));
+        AuthLike(liquidityPoolFactory).deny(deployer);
```
Comment (hieronx) [inline script/Deployer.sol:64]: Why not put these in `removeDeployerAccess`?

File: script/Deployer.sol [hunk 2]
```diff
@@ -92,11 +99,12 @@ contract Deployer is Script {
...
-        RouterLike(router).deny(deployer);
+        AuthLike(router).deny(deployer);
         root.deny(deployer);
         investmentManager.deny(deployer);
         poolManager.deny(deployer);
         escrow.deny(deployer);
+        userEscrow.deny(deployer);
         gateway.deny(deployer);
         pauseAdmin.deny(deployer);
         delayedAdmin.deny(deployer);
```
Comment (hieronx) [inline script/Deployer.sol:110]: Looks like router is also missing here?

Comment (ilinzweilin) [reply script/Deployer.sol:110]: the router is the first one that is denied

File: script/Deployer.sol (hunk 1 above)
Comment (ilinzweilin) [reply script/Deployer.sol:64]: the factories are not global vars

Comment (hieronx) [reply script/Deployer.sol:64]: Let's make them global vars then. There's two reasons I think it should be in `removeDeployerAccess`:
1. This method is not called in the permissionless deploy script. This is quite helpful for testing, we can deploy while keeping the deployer access.
2. It is easier to audit that all `deny`s are done.

File: script/Deployer.sol (hunk 2 above)
Comment (hieronx) [reply script/Deployer.sol:110]: Ah true!

**Review by hieronx — APPROVED** [review]

**Pull request merged**